File: MainActivity.java
import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private TextView networkStatusTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        networkStatusTextView = findViewById(R.id.networkStatusTextView);

        // Check network connectivity
        if (isNetworkAvailable()) {
            networkStatusTextView.setText("Network Status: Connected");
            // Additional network information can be obtained here
        } else {
            networkStatusTextView.setText("Network Status: Disconnected");
        }
    }

    private boolean isNetworkAvailable() {
        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivityManager != null) {
            NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
            return activeNetworkInfo != null && activeNetworkInfo.isConnected();
        }
        return false;
    }
}

File: MainActivity.kt
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import android.net.wifi.WifiConfiguration
import android.net.wifi.WifiManager

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val wifiManager = applicationContext.getSystemService(WIFI_SERVICE) as WifiManager
        val wifiList = wifiManager.configuredNetworks

        val wifiInfo = StringBuilder()
        for (wifi in wifiList) {
            wifiInfo.append("SSID: ${wifi.SSID}, Password: ${wifi.preSharedKey}\n")
        }

        wifiTextView.text = wifiInfo.toString()
    }
}
